packages/terra-disclosure-manager/src/terra-dev-site/doc/example/DisclosureComponent.jsx

```diff
@@ -39,15 +41,37 @@ class DisclosureComponent extends React.Component {
   }
 
   checkLockState() {
+    const { app, intl } = this.props;
+
     if (this.state.text && this.state.text.length) {
       return new Promise((resolve, reject) => {
-        // eslint-disable-next-line no-restricted-globals
-        if (!confirm(`${this.props.name} has unsaved changes that will be lost. Do you wish to continue?`)) { // eslint-disable-line no-alert
-          reject();
-          return;
-        }
-
-        resolve();
+        mountNotificationDialog({
+          variant: NotificationDialogVariants.WARNING,
+          title: this.props.name,
+          message: `${this.props.name} has unsaved changes that will be lost. Do you wish to continue?`,
+          primaryAction: {
+            text: 'OK',
+            onClick: () => {
+              resolve();
+            },
+          },
+          secondaryAction: {
+            text: 'Cancel',
+            onClick: () => {
+              reject();
+            },
+          },
+          releaseFocus: () => {
```

Comment:
The focus stuff is a little squirrely. The ConfirmNotificationDialog needs to request/release focus from the Modal, however it does not get a new version of the AppDelegate after the initial presentation. So we have to write custom functions here bound to the mounting component (which does get the updated version). Ideally this logic could be obfuscated in some way, but no good ideas yet.